src/isl/util.rs

```diff
@@ -240,13 +242,44 @@ impl Range {
         })
     }
 
+    pub fn validate_precision_range(precision_range: &Range) -> IonSchemaResult<Range> {
+        // minimum precision must be greater than or equal to 1
+        // for more information: https://amzn.github.io/ion-schema/docs/spec.html#precision
+        match precision_range {
+            Range::IntegerNonNegative(min_value, max_value) => match min_value {
+                RangeBoundaryValue::Max => Ok(precision_range.to_owned()),
+                RangeBoundaryValue::Min => {
+                    return Range::range(
```

Comment:
To fix [clippy's warning](https://github.com/amzn/ion-schema-rust/pull/83/files#annotation_3574931928):
```suggestion
                    Range::range(
```